app/controllers/topics_controller.rb

```diff
@@ -1077,11 +1082,30 @@ def reset_new
       topic_scope = topic_scope.where(id: topic_ids)
     end
 
-    dismissed_topic_ids =
-      TopicsBulkAction.new(current_user, topic_scope.pluck(:id), type: "dismiss_topics").perform!
-    TopicTrackingState.publish_dismiss_new(current_user.id, topic_ids: dismissed_topic_ids)
+    dismissed_topic_ids = []
+    dismissed_post_topic_ids = []
+    if !current_user.new_new_view_enabled? || params[:dismiss_topics]
+      dismissed_topic_ids =
+        TopicsBulkAction.new(current_user, topic_scope.pluck(:id), type: "dismiss_topics").perform!
+      TopicTrackingState.publish_dismiss_new(current_user.id, topic_ids: dismissed_topic_ids)
+    end
 
-    render body: nil
+    if params[:dismiss_posts]
+      dismissed_post_topic_ids =
+        TopicsBulkAction.new(current_user, topic_scope.pluck(:id), type: "dismiss_posts").perform!
+    end
+
+    topic_ids = dismissed_topic_ids.concat(dismissed_post_topic_ids)
+    if params[:untrack]
+      TopicsBulkAction.new(
+        current_user,
+        topic_ids,
+        type: "change_notification_level",
+        notification_level_id: NotificationLevels.topic_levels[:regular],
+      ).perform!
+    end
```

Comment:
The way "untracking" currently works in the old /new view is that if the user decides they want to untrack unread topics, we do a `change_notification_level` operation instead of `dismiss_posts`, but not both:

https://github.com/discourse/discourse/blob/20982ef875a149132be3ea04265a81377a5c6d76/app/assets/javascripts/discourse/app/mixins/bulk-topic-selection.js#L41C1-L47

I think we should do the same here and move the `if params[:untrack]` block inside the `if params[:dismiss_posts]` block and perform `dismiss_posts` if `untrack` is false, otherwise perform `change_notification_level`.